harness/tests/integration_cases/test_raft.rs

```diff
@@ -212,6 +212,121 @@ fn test_progress_update() {
     }
 }
 
+#[test]
+fn test_progress_committed() {
```

Comment:
I think you can use `Network` to test the commit index change. It should validate following case:
1. Log rejection should not change commit index
2. Successfully Append should update commit index
3. Heartbeat should update commit index
4. Commit index won't step backward.

Reply:
I still do not know when will we get the smaller commit index

Reply:
For example, leader sends two `MsgAppend` to follower and gets `MsgAppendResponse` in reverse order.